File: services/device/src/main/java/com/example/iter/device/dto/response/EquipmentImageResponse.java
package com.example.iter.device.dto.response;

import com.example.iter.common.image.CaptureView;
import com.example.iter.device.domain.entity.EquipmentImage;

public record EquipmentImageResponse(
        Long id,
        String imageUrl,
        CaptureView captureView,
        int sortOrder,
        boolean thumbnail
) {
    public static EquipmentImageResponse from(EquipmentImage image, String resolvedImageUrl) {
        return new EquipmentImageResponse(
                image.getId(),
                resolvedImageUrl,
                image.getCaptureView(),
                image.getSortOrder(),
                image.isThumbnail()
        );
    }
}

File: services/device/src/main/kotlin/com/example/iter/device/dto/response/EquipmentImageResponse.kt
package com.example.iter.device.dto.response

import com.example.iter.device.domain.entity.EquipmentImage

data class EquipmentImageResponse(
    val id: Long?,
    val imageUrl: String,
    val sortOrder: Int,
    val thumbnail: Boolean,
) {
    companion object {
        // 자바에서 EquipmentImageResponse.from(...) 으로 부르므로 @JvmStatic 이 필요하다.
        @JvmStatic
        fun from(image: EquipmentImage, resolvedImageUrl: String) = EquipmentImageResponse(
            id = image.id,
            imageUrl = resolvedImageUrl,
            sortOrder = image.sortOrder,
            thumbnail = image.thumbnail,
        )
    }
}
